src/signTxCatalystRegistration.c

```diff
@@ -0,0 +1,576 @@
+#include "signTxCatalystRegistration.h"
+#include "state.h"
+#include "uiHelpers.h"
+#include "signTxUtils.h"
+#include "uiScreens.h"
+#include "auxDataHashBuilder.h"
+#include "txHashBuilder.h"
+#include "textUtils.h"
+#include "bufView.h"
+#include "securityPolicy.h"
+#include "messageSigning.h"
+
+// we want to distinguish the two state machines to avoid potential confusion:
+// ctx / subctx
+// stage / state
+// from ctx, we only make the necessary parts available to avoid mistaken overwrites
+static ins_sign_tx_context_t* ctx = &(instructionState.signTxContext);
```

Comment:
It's actually used in two functions in multiple places albeit for the same thing - dealing with the auxDataHash stored in the top-level instruction state in order to propagate it to signTx function which is responsible of including it in the tx body in a later step. So I'm not sure if it would be practical to hide it within the functions where it's being used.

But I agree it makes sense to do the initialization of the subctx the way it's done in the operator app, in order to prevent mixups of contexts